### PR TITLE
Generic handler cmd commands

### PR DESCRIPTION
**Pull request by mpetrun5**
## Description

Generic handler deploy and interaction CLI commands

## Related Issue Or Context

Closes: #100 

## How Has This Been Tested? Testing details.

Tested all commands on chainbridge-core-example repo manually.
Still need to create generic transfer between networks, but it will
be easier to setup that with code.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation

## Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have ensured that all acceptance criteria (or expected behavior) from issue are met
- [x] I have updated the documentation locally and in chainbridge-docs.
- [x] I have added tests to cover my changes.
- [x] I have ensured that all the checks are passing and green, I've signed the CLA bot

### CODE REVIEW

File: chains/evm/calls/bridge.go [hunk 1]
```diff
@@ -37,6 +37,24 @@ func PrepareAdminSetResourceInput(handler common.Address, rId [32]byte, addr com
 	return input, nil
 }
 
+func PrepareAdminSetGenericResourceInput(
+	handler common.Address,
+	rId [32]byte,
```
Comment (nmlinaric) [inline chains/evm/calls/bridge.go:42]: I think you need to rebase to `develop`, this is now `resourceID`

File: chains/evm/cli/bridge/register-generic-resource.go [hunk 2]
```diff
@@ -9,27 +17,39 @@ var registerGenericResourceCmd = &cobra.Command{
...
 	Long:  "Register a resource ID with a contract address for a generic handler",
-	Run:   registerGenericResource,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		txFabric := evmtransaction.NewTransaction
+		return RegisterGenericResourceCmd(cmd, args, txFabric)
+	},
+}
+
+func BindRegisterGenericResourceCmdFlags(cmd *cobra.Command) {
+	cmd.Flags().String("handler", "", "handler contract address")
```
[the remark this answers is not in the document]
Comment (mpetrun5) [reply chains/evm/cli/bridge/register-generic-resource.go:27]: @P1sar The old `cb-sol-cli` has for these flags default values which are basically hardcoded for easier configuration when running locally. The go cli up to this point hasn't really been doing that. 

Are we planning to introduce the same default value logic or we should make all those flags required?

[the remark this answers is not in the document]
Comment (mpetrun5) [reply chains/evm/cli/bridge/register-generic-resource.go:27]: :+1: 

When we do cmd rewrite we should check for all flags that previously had default values and mark them as required @tcar121293

[the remark this answers is not in the document]
Comment (MakMuftic) [reply chains/evm/cli/bridge/register-generic-resource.go:27]: I will add a note for this in the rewrite issue itself

File: chains/evm/cli/bridge/register-generic-resource.go [hunk 3]
```diff
@@ -9,27 +17,56 @@ var registerGenericResourceCmd = &cobra.Command{
...
+	cmd.Flags().String("resourceId", "", "resource ID to query")
+	cmd.Flags().String("bridge", "", "bridge contract address")
+	cmd.Flags().String("target", "", "contract address to be registered")
+	cmd.Flags().String("deposit", "0x00000000", "deposit function signature")
+	cmd.Flags().String("execute", "0x00000000", "execute proposal function signature")
+	cmd.Flags().Bool("hash", false, "treat signature inputs as function prototype strings, hash and take the first 4 bytes")
+
+	err := cmd.MarkFlagRequired("handler")
```
Comment (MakMuftic) [inline chains/evm/cli/bridge/register-generic-resource.go:35]: @tcar121293 Maybe this is something that can be wrapped in a separate function (in your rewrite), as we will have a lot of these snips trough out the codebase.

**Review by nmlinaric — APPROVED** [review]

**Pull request merged**